### PR TITLE
HOSTEDCP-954: Remove ec2:ReleaseAddress

### PR DESCRIPTION
**Pull request by fahlmant**
**What this PR does / why we need it**:

Checking if ec2:ReleaseAddress is needed for NodePool management polcy or not 

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes https://issues.redhat.com/browse/HOSTEDCP-954

**Checklist**
- [x ] Subject and description added to both, commit and PR.
- [ x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/infra/aws/iam.go [hunk 1]
```diff
@@ -211,7 +211,6 @@ const (
         "ec2:ModifyInstanceAttribute",
         "ec2:ModifyNetworkInterfaceAttribute",
         "ec2:ModifySubnetAttribute",
-        "ec2:ReleaseAddress",
```
Comment (enxebre) [inline cmd/infra/aws/iam.go:214]: can you please also remove AllocateAddress
and remove them from the API docs https://github.com/openshift/hypershift/blob/main/api/v1beta1/hostedcluster_types.go#L1285?

**Comment by fahlmant** [comment]
area/hypershift-operator

**Review by jharrington22 — COMMENTED** [review]
@enxebre what is the infra cmd? It looks like its referencing the [ReleaseAddress](https://github.com/openshift/hypershift/blob/main/cmd/infra/aws/destroy.go#L644-L653) API?

**Comment by enxebre** [comment]
@fahlmant you need to `make update` to update automation with the API comment and get CI happy

**Comment by enxebre** [comment]
>@enxebre what is the infra cmd? It looks like its referencing the [ReleaseAddress](https://github.com/openshift/hypershift/blob/main/cmd/infra/aws/destroy.go#L644-L653) API?

@jharrington22 hypershift provides a helper CLI for devs to bring their infra before creating a HostedCluster.
In that infra topology we create nat gateways with EIPs.

**Comment by jharrington22** [comment]
> > @enxebre what is the infra cmd? It looks like its referencing the [ReleaseAddress](https://github.com/openshift/hypershift/blob/main/cmd/infra/aws/destroy.go#L644-L653) API?
> 
> @jharrington22 hypershift provides a helper CLI for devs to bring their infra before creating a HostedCluster. In that infra topology we create nat gateways with EIPs.
> 
> That's separate from the iam policies that the operators we run might need.

I thought that might be the case just making sure. Thanks @enxebre

**Pull request merged**